samples/net/mqtt/src/modules/transport/credentials_provision/credentials_provision.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <modem/modem_key_mgmt.h>
+
+static const unsigned char ca_certificate[] = {
+#include "ca-cert.pem"
+};
+
+static const unsigned char device_certificate[] = {
+#include "client-cert.pem"
+};
+
+static const unsigned char private_key[] = {
+#include "private-key.pem"
+};
+
+int credentials_provision(void)
```

Comment:
What is the rationale for having the same API implemented, albeit for native TLS, in the MQTT helper? Can we pull out that functionality out of MQTT helper and put it here?

Note, that I think we need a single API for managing credentials rather independent on the device. This is something I would like to address next but for now I'd at least provisioning done in one place.

https://github.com/nrfconnect/sdk-nrf/blob/bc36642b0479a9c90492c6f10ddd36d139b0aa27/subsys/net/lib/mqtt_helper/mqtt_helper.c#L147

Reply:
> What is the rationale for having the same API implemented, albeit for native TLS, in the MQTT helper? Can we pull out that functionality out of MQTT helper and put it here?
> 
> Note, that I think we need a single API for managing credentials rather independent on the device. This is something I would like to address next but for now I'd at least provisioning done in one place.
> 
> https://github.com/nrfconnect/sdk-nrf/blob/bc36642b0479a9c90492c6f10ddd36d139b0aa27/subsys/net/lib/mqtt_helper/mqtt_helper.c#L147

A single API for managing credenials sounds like a good idea. The intention is that AWS IoT, Azure IoT Hub, and possibly nRF Cloud lib will all use the MQTT helper library. We have on commit tests for the corresponding samples that uses TLS + Native Posix (also for ATv2). So to prevent duplicating code across samples to provision credentials it was put into MQTT helper.

We could place the Modem key MGMT stuff into MQTT helper for now and pull it out later when such an API is to be made. It would also be nice if modem key management could hook into the tls_credentials API so that we don't need to call different APIs as well.